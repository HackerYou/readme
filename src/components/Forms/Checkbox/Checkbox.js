import React from 'react';
import PropTypes from 'prop-types';
import keygen from 'keygenerator';

const Checkbox = ({ id, handleChange, labelText }) => {
    return (
        <span>
            <label htmlFor={id}>{labelText}</label>
            <input onChange={handleChange} id={id} type="checkbox" />
        </span>
    );
};

Checkbox.defaultProps = {
    labelText: '',
    handleChange: null,
    id: keygen._(),
};

Checkbox.propTypes = {
    id: PropTypes.string.isRequired,
    handleChange: PropTypes.func,
    labelText: PropTypes.string,
};

export default Checkbox;
