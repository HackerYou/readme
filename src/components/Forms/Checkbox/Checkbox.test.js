import React from 'react';
import { shallow, mount } from 'enzyme';
import Checkbox from './Checkbox';
import renderer from 'react-test-renderer';

test('It renders the checkbox component', () => {
    const props = {
        labelText: "Title",
        id: "8675309",
    };
    const component = renderer.create(<Checkbox {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});