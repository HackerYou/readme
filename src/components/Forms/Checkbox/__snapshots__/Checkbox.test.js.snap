// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders the checkbox component 1`] = `
<span>
  <label
    htmlFor="8675309"
  >
    Title
  </label>
  <input
    id="8675309"
    onChange={null}
    type="checkbox"
  />
</span>
`;
